account_bank_statement_import_adyen/models/account_bank_statement_import.py

```diff
@@ -0,0 +1,135 @@
+            date=fields.Date.from_string(row[6]),
+            amount=self.balance(row),
+            note='%s %s %s %s' % (row[2], row[3], row[4], row[21]),
+            name="%s" % (row[3] or row[4] or row[9]),
+        )
+        statement['transactions'].append(transaction)
+
+    @api.model
+    def import_adyen_xlsx(self, data_file):
+        statements = []
+        statement = None
+        headers = False
+        fees = 0.0
+        balance = 0.0
+        payout = 0.0
+        statement_id = None
+
+        with BytesIO() as buf:
+            buf.write(data_file)
+            try:
+                sheet = load_workbook(buf)._sheets[0]
+            except BadZipfile as e:
+                raise ValueError(e)
+            for row in sheet.rows:
+                row = [cell.value for cell in row]
+                if len(row) != 31:
+                    raise ValueError(
+                        'Not an Adyen statement. Unexpected row length %s '
+                        'instead of 31' % len(row))
+                if not row[1]:
+                    continue
+                if not headers:
+                    if row[1] != 'Company Account':
+                        raise ValueError(
+                            'Not an Adyen statement. Unexpected header "%s" '
+                            'instead of "Company Account"', row[1])
+                    headers = True
+                    continue
+                if not statement:
+                    statement = {'transactions': []}
+                    statements.append(statement)
+                    statement_id = '%s %s/%s' % (
+                        row[2], row[6].strftime('%Y'), int(row[23]))
+                    currency_code = row[14]
+                    merchant_id = row[2]
+                    statement['name'] = '%s %s/%s' % (
+                        row[2], row[6].year, row[23])
+                date = fields.Date.from_string(row[6])
+                if not statement.get('date') or statement.get('date') > date:
+                    statement['date'] = date
+
+                row[8] = row[8].strip()
+                if row[8] == 'MerchantPayout':
+                    payout -= self.balance(row)
+                else:
+                    balance += self.balance(row)
+                self.import_adyen_transaction(statement, statement_id, row)
+                fees += sum(
+                    row[i] if row[i] else 0.0
+                    for i in (17, 18, 19, 20))
+
+        if not headers:
+            raise ValueError(
+                'Not an Adyen statement. Did not encounter header row.')
+
+        if fees:
+            transaction_id = str(len(statement['transactions'])).zfill(4)
+            transaction = dict(
+                unique_import_id=statement_id + transaction_id,
+                date=max(t['date'] for t in statement['transactions']),
+                amount=-fees,
+                name='Commission, markup etc. batch %s' % (int(row[23])),
```

Comment:
row[23] not necessary conversion to int, is a problem if row[23] is a text

`name='Commission, markup etc. batch %s' % (row[23])`